### PR TITLE
[[renesas/drivers] Fix error code issues 6183.Add hardware i2c driver.

### PR DESCRIPTION
**Pull request by ShermanShao**
## 拉取/合并请求描述：(PR description)

[
[[renesas/drv_wdt.c] Fix error code issues [6183](https://github.com/RT-Thread/rt-thread/issues/6183)
[renesas/drv_i2c.c] Add hardware i2c driver
]

以下的内容不应该在提交PR时的message修改，修改下述message，PR会被直接关闭。请在提交PR后，浏览器查看PR并对以下检查项逐项check，没问题后逐条在页面上打钩。
The following content must not be changed in the submitted PR message. Otherwise, the PR will be closed immediately. After submitted PR, please use a web browser to visit PR, and check items one by one, and ticked them if no problem.

### 当前拉取/合并请求的状态 Intent for your PR

必须选择一项 Choose one (Mandatory):

- [ ] 本拉取/合并请求是一个草稿版本 This PR is for a code-review and is intended to get feedback
- [x] 本拉取/合并请求是一个成熟版本 This PR is mature, and ready to be integrated into the repo

### 代码质量 Code Quality：

我在这个拉取/合并请求中已经考虑了 As part of this pull request, I've considered the following:

- [ ] 已经仔细查看过代码改动的对比 Already check the difference between PR and old code
- [ ] 代码风格正确，包括缩进空格，命名及其他风格 Style guide is adhered to, including spacing, naming and other styles
- [ ] 没有垃圾代码，代码尽量精简，不包含`#if 0`代码，不包含已经被注释了的代码 All redundant code is removed and cleaned up
- [ ] 所有变更均有原因及合理的，并且不会影响到其他软件组件代码或BSP All modifications are justified and not affect other components or BSP
- [ ] 对难懂代码均提供对应的注释 I've commented appropriately where code is tricky
- [ ] 本拉取/合并请求代码是高质量的 Code in this PR is of high quality
- [ ] 本拉取/合并使用[formatting](https://github.com/mysterywolf/formatting)等源码格式化工具确保格式符合[RT-Thread代码规范](../documentation/contribution_guide/coding_style_cn.md) This PR complies with [RT-Thread code specification](../documentation/contribution_guide/coding_style_en.txt)

### CODE REVIEW

File: bsp/renesas/libraries/HAL_Drivers/drv_wdt.c [hunk 1]
```diff
@@ -39,40 +40,46 @@ static rt_err_t wdt_control(rt_watchdog_t *wdt, int cmd, void *arg)
         if (R_WDT_Refresh(&g_wdt_ctrl) != FSP_SUCCESS)
         {
             LOG_E("watch dog keepalive fail.");
+            ret =  -RT_ERROR;
         }
+        ret = RT_EOK;
```
Comment (mysterywolf) [inline bsp/renesas/libraries/HAL_Drivers/drv_wdt.c:45]: 这个逻辑有问题吧 设置为-的错误码之后立刻就又变成了RT_EOK

**Review by mysterywolf — CHANGES_REQUESTED** [review]

File: bsp/renesas/libraries/HAL_Drivers/drv_i2c.c [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+                if(FSP_ERR_TRANSFER_ABORTED == err)
+                {
+                    LOG_E("** POWER_CTL reg I2C write failed!!! ** \n");
+                }
+            }
+            /* handle error */
+            else
+            {
+                /* Write API returns itself is not successful */
+                LOG_E("** R_IIC_MASTER_Write API failed ** \n");
```
Comment (mysterywolf) [inline bsp/renesas/libraries/HAL_Drivers/drv_i2c.c:132]: 此处 如果出现错误了 是否还有必要继续往下执行for循环？

File: bsp/renesas/libraries/HAL_Drivers/drv_i2c.c [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+    for (i = 0; i < num; i++)
+    {
+
+        if (msg[i].flags & RT_I2C_NO_START)
+        {
+            restart = true;
+        }
+        if (msg[i].flags & RT_I2C_ADDR_10BIT)
+        {
+            LOG_E("10Bit not support!\n");
```
Comment (mysterywolf) [inline bsp/renesas/libraries/HAL_Drivers/drv_i2c.c:89]: LOG_X 无需加星号或者\r\n LOG_X会自动帮忙换行

**Review by mysterywolf — CHANGES_REQUESTED** [review]

File: bsp/renesas/libraries/HAL_Drivers/drv_wdt.c (hunk 1 above)
Comment (ShermanShao) [reply bsp/renesas/libraries/HAL_Drivers/drv_wdt.c]: 是的 已修复

File: bsp/renesas/libraries/HAL_Drivers/drv_i2c.c (hunk 2 above)
Comment (ShermanShao) [reply bsp/renesas/libraries/HAL_Drivers/drv_i2c.c]: 感觉还是直接退出吧 已修复

File: bsp/renesas/libraries/HAL_Drivers/drv_i2c.c (hunk 3 above)
Comment (ShermanShao) [reply bsp/renesas/libraries/HAL_Drivers/drv_i2c.c]: 已修复

File: bsp/renesas/libraries/HAL_Drivers/drv_i2c.c [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+    {
+
+        if (msg[i].flags & RT_I2C_NO_START)
+        {
+            restart = true;
+        }
+        if (msg[i].flags & RT_I2C_ADDR_10BIT)
+        {
+            LOG_E("10Bit not support!");
+            return -1;
```
Comment (mysterywolf) [inline bsp/renesas/libraries/HAL_Drivers/drv_i2c.c:90]: 这个位置 错误不能返回-1 只能返回0 rt_size_t 是无符号类型

**Review by mysterywolf — APPROVED** [review]

**Review by Guozhanxin — APPROVED** [review]

**Pull request merged**